### DISCUSSION TITLE
[mkdocs-material] ModuleNotFoundError: No module named 'material.plugins'

### DISCUSSION THREAD

**Discussion by MathieuAibt** [squidfunk/mkdocs-material#4646] [Q&A] [answered]
Hi @squidfunk,

First of all, thanks in advance for this wonderful piece of package you're developing/maintaining ;)
However, I can't seem to install it correctly and I'm now bumping into following `ModuleNotFoundError: No module named 'material.plugins'` error that was apparently solved in previous commit (https://github.com/squidfunk/mkdocs-material/tree/f139b54c61b2df4512f15149cba2c222475e2bc9/material/plugins).

Having downloaded the 8.5.9 version, I should benefit from these fixes.
Would you, by any chance, have some hints on how to fix this?

Below is a `conda info` report. Please do not hesitate to ask for any additional information.
```
(mkdocs_only) C:\>conda info

     active environment : mkdocs_only
    active env location : C:\Users\<my_login>\Miniconda3\envs\mkdocs_only
            shell level : 2
       user config file : C:\Users\<my_login>\.condarc
 populated config files : C:\Users\<my_login>\.condarc
          conda version : 22.9.0
    conda-build version : not installed
         python version : 3.9.13.final.0
       virtual packages : __win=0=0
                          __archspec=1=x86_64
       base environment : C:\Users\<my_login>\Miniconda3  (writable)
      conda av data dir : C:\Users\<my_login>\Miniconda3\etc\conda
  conda av metadata url : None
           channel URLs : ...
          package cache : C:\Users\<my_login>\Miniconda3\pkgs
                          C:\Users\<my_login>\.conda\pkgs
                          C:\Users\<my_login>\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\<my_login>\Miniconda3\envs
                          C:\Users\<my_login>\.conda\envs
                          C:\Users\<my_login>\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/22.9.0 requests/2.28.1 CPython/3.9.13 Windows/10 Windows/10.0.19042
          administrator : False
             netrc file : None
           offline mode : False
```
and the `pip list` report:
```
(mkdocs_only) C:\>pip list
Package                    Version
-------------------------- ---------
certifi                    2022.9.24
charset-normalizer         2.1.1
click                      8.1.3
colorama                   0.4.6
ghp-import                 2.1.0
griffe                     0.24.0
idna                       3.4
importlib-metadata         5.0.0
Jinja2                     3.1.2
Markdown                   3.3.7
MarkupSafe                 2.1.1
mergedeep                  1.3.4
mkdocs                     1.4.2
mkdocs-autorefs            0.4.1
mkdocs-material            8.5.9
mkdocs-material-extensions 1.1
mkdocstrings               0.19.0
mkdocstrings-python        0.7.1
packaging                  21.3
pip                        22.3.1
Pygments                   2.13.0
pymdown-extensions         9.8
pyparsing                  3.0.9
python-dateutil            2.8.2
PyYAML                     6.0
pyyaml_env_tag             0.1
requests                   2.28.1
setuptools                 65.5.1
six                        1.16.0
urllib3                    1.26.12
watchdog                   2.1.9
wheel                      0.38.4
zipp                       3.10.0
```

and also the step-by-step process I used to install my conda environment:
```
conda create --name mkdocs_only python=3.9 
conda activate mkdocs_only  

pip install --upgrade pip
pip install mkdocs==1.4.2
pip install mkdocstrings-python==0.7.1  
pip install local_dir\mkdocs_material-8.5.9-py3-none-any.whl
```

_Note that I used a modified wheel in order to remove `kaspersky.svg` file (leading to PermissionError due to my company antivirus policy)_

Also, be aware that I tried with and without constraining mkdocs and mkdocsstring-python version with similar outcomes.

Thanks in advance.
Regards,
Mathieu

**Comment by squidfunk** [comment, marked as answer]
Thanks for reporting. We're not officially supporting `conda`. Does the same happen when you install from [`pip`](https://squidfunk.github.io/mkdocs-material/getting-started/#with-pip) or via [`git`](https://squidfunk.github.io/mkdocs-material/getting-started/#with-git)? Also note that when you use a modified wheel, we can't offer official debugging support. We just don't have the time to.

Maybe another user can help here.

**Reply by MathieuAibt** [reply]
Thanks for answering. As for your questions.

- I am using a conda environment. **However**, as displayed in my initial message, the mkdocs-related packages were installed using `pip`.
- As for the wheel, I downloaded the latest version (8.5.9 at that time) from https://pypi.org/project/mkdocs-material/#files
- Then, using 7-zip, I dived into subfolders until reaching `kaspersky.svg` file ('mkdocs_material-8.5.9\\material\\.icons\\simple\\kaspersky.svg')
- And I just deleted the file, leaving the rest of the wheel untouched
- This file was then stored in `local_dir`
- and installed using `pip install local_dir\mkdocs_material-8.5.9-py3-none-any.whl`

I guess this might have little to no impact on the rest of the wheel/package/install.
Am I right ?

Don't hesitate to ask for any additional information that might be necessary for you to further help me.
Thanks
Mathieu

**Reply by squidfunk** [reply]
The fact is: I can't reproduce your issue. Everything works as expected. Maybe you can upgrade to 8.5.10 and check if the error is still there. Other than that, there's very likely a problem with your setup. Try using a virtual environment to start clean.

```
squidfunk@squidfunk:~/Desktop > cd test

squidfunk@squidfunk:~/Desktop/test > virtualenv venv
/usr/local/lib/python3.8/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
created virtual environment CPython3.8.14.final.0-64 in 663ms
  creator CPython3Posix(dest=/Users/squidfunk/Desktop/test/venv, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/squidfunk/Library/Application Support/virtualenv)
    added seed packages: pip==22.3, setuptools==65.5.0, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

squidfunk@squidfunk:~/Desktop/test > . ./venv/bin/activate                          
                          
(venv) squidfunk@squidfunk:~/Desktop/test > pip install mkdocs-material       
Collecting mkdocs-material
  Using cached mkdocs_material-8.5.10-py3-none-any.whl (7.5 MB)
Collecting pygments>=2.12
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Collecting jinja2>=3.0.2
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting markdown>=3.2
  Using cached Markdown-3.4.1-py3-none-any.whl (93 kB)
Collecting pymdown-extensions>=9.4
  Using cached pymdown_extensions-9.8-py3-none-any.whl (218 kB)
Collecting mkdocs>=1.4.0
  Using cached mkdocs-1.4.2-py3-none-any.whl (3.7 MB)
Collecting mkdocs-material-extensions>=1.1
  Using cached mkdocs_material_extensions-1.1-py3-none-any.whl (7.9 kB)
Collecting requests>=2.26
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.1-cp38-cp38-macosx_10_9_x86_64.whl (13 kB)
Collecting importlib-metadata>=4.4
  Using cached importlib_metadata-5.0.0-py3-none-any.whl (21 kB)
Collecting click>=7.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting mergedeep>=1.3.4
  Using cached mergedeep-1.3.4-py3-none-any.whl (6.4 kB)
Collecting markdown>=3.2
  Using cached Markdown-3.3.7-py3-none-any.whl (97 kB)
Collecting packaging>=20.5
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting pyyaml-env-tag>=0.1
  Using cached pyyaml_env_tag-0.1-py3-none-any.whl (3.9 kB)
Collecting watchdog>=2.0
  Using cached watchdog-2.1.9-cp38-cp38-macosx_10_9_x86_64.whl (87 kB)
Collecting ghp-import>=1.0
  Using cached ghp_import-2.1.0-py3-none-any.whl (11 kB)
Collecting pyyaml>=5.1
  Using cached PyYAML-6.0-cp38-cp38-macosx_10_9_x86_64.whl (192 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting python-dateutil>=2.8.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting zipp>=0.5
  Using cached zipp-3.10.0-py3-none-any.whl (6.2 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: zipp, watchdog, urllib3, six, pyyaml, pyparsing, pygments, mkdocs-material-extensions, mergedeep, MarkupSafe, idna, click, charset-normalizer, certifi, requests, pyyaml-env-tag, python-dateutil, packaging, jinja2, importlib-metadata, markdown, ghp-import, pymdown-extensions, mkdocs, mkdocs-material
Successfully installed MarkupSafe-2.1.1 certifi-2022.9.24 charset-normalizer-2.1.1 click-8.1.3 ghp-import-2.1.0 idna-3.4 importlib-metadata-5.0.0 jinja2-3.1.2 markdown-3.3.7 mergedeep-1.3.4 mkdocs-1.4.2 mkdocs-material-8.5.10 mkdocs-material-extensions-1.1 packaging-21.3 pygments-2.13.0 pymdown-extensions-9.8 pyparsing-3.0.9 python-dateutil-2.8.2 pyyaml-6.0 pyyaml-env-tag-0.1 requests-2.28.1 six-1.16.0 urllib3-1.26.12 watchdog-2.1.9 zipp-3.10.0

(venv) squidfunk@squidfunk:~/Desktop/test > mkdocs new .                            
INFO     -  Writing config file: ./mkdocs.yml
INFO     -  Writing initial docs: ./docs/index.md

(venv) squidfunk@squidfunk:~/Desktop/test > echo "theme:\n name: material" >> mkdocs.yml      
       
(venv) squidfunk@squidfunk:~/Desktop/test > mkdocs build
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: /Users/squidfunk/Desktop/test/site
INFO     -  Documentation built in 0.31 seconds
```

**Reply by MathieuAibt** [reply]
Thanks again.
I found something interesting...

# Changes I made
- Based on one error I encountered a little while ago, regarding setuptools file being corrupted,
- and witnessing that I was using a newer version as your (65.5.1),
- I decided to delete all setuptools files/directories from: _[EDIT: wrong lead...see in following answers]_

The results is that, when creating my blank conda environment, the `setuptools-65.5.1` files where downloaded as brand new files.
I also used python 3.8 in order to match your example.

# Results : working
From there, I did pretty much as you did and it worked.
_Note that I had to make a change inside mkdocs.yml since "theme:\n name: material" was added as is, without any evaluation_

# Way forward

> **--QUESTION--** Would you happen to know how to force a specific version of setuptools to be re-downloaded and installed ?
(I had a quick look but did not find that much yet)

> **--TODO--** I will do the same with python 3.9 and report the status asap

# Prompt below
```
(base) C:\Users\<my_login>\Desktop\p38>conda create --name mkdocs_p38 python=3.8
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\<my_login>\Miniconda3\envs\mkdocs_p38

  added / updated specs:
    - python=3.8


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    setuptools-65.5.1          |     pyhd8ed1ab_0         731 KB  defaults
    ------------------------------------------------------------
                                           Total:         731 KB

The following NEW packages will be INSTALLED:

  bzip2              anaconda-org/conda-forge/win-64::bzip2-1.0.8-h8ffe710_4 None
  ca-certificates    anaconda-org/conda-forge/win-64::ca-certificates-2022.9.24-h5b45459_0 None
  libffi             anaconda-org/conda-forge/win-64::libffi-3.4.2-h8ffe710_5 None
  libsqlite          anaconda-org/conda-forge/win-64::libsqlite-3.39.4-hcfcfb64_0 None
  libzlib            anaconda-org/conda-forge/win-64::libzlib-1.2.13-hcfcfb64_4 None
  openssl            anaconda-org/conda-forge/win-64::openssl-3.0.7-hcfcfb64_0 None
  pip                anaconda-org/conda-forge/noarch::pip-22.3.1-pyhd8ed1ab_0 None
  python             anaconda-org/conda-forge/win-64::python-3.8.13-hcf16a7b_0_cpython None
  setuptools         anaconda-org/conda-forge/noarch::setuptools-65.5.1-pyhd8ed1ab_0 None
  sqlite             anaconda-org/conda-forge/win-64::sqlite-3.39.4-hcfcfb64_0 None
  tk                 anaconda-org/conda-forge/win-64::tk-8.6.12-h8ffe710_0 None
  ucrt               anaconda-org/conda-forge/win-64::ucrt-10.0.22621.0-h57928b3_0 None
  vc                 anaconda-org/conda-forge/win-64::vc-14.3-h3d8a991_9 None
  vs2015_runtime     anaconda-org/conda-forge/win-64::vs2015_runtime-14.32.31332-h1d6e394_9 None
  wheel              anaconda-org/conda-forge/noarch::wheel-0.38.4-pyhd8ed1ab_0 None
  xz                 anaconda-org/conda-forge/win-64::xz-5.2.6-h8d14728_0 None


Proceed ([y]/n)? y


Downloading and Extracting Packages
setuptools-65.5.1    | 731 KB    | ################################################################################################################# | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate mkdocs_p38
#
# To deactivate an active environment, use
#
#     $ conda deactivate

Retrieving notices: ...working... done

(base) C:\Users\<my_login>\Desktop\p38>conda activate mkdocs_p38

(mkdocs_p38) C:\Users\<my_login>\Desktop\p38>pip install S:\CODE_INDUS\_exchange\Dev_Setup\mkdocs_material-8.5.10-py3-none-any.whl
Looking in indexes: https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/simple
Processing s:\code_indus\_exchange\dev_setup\mkdocs_material-8.5.10-py3-none-any.whl
Collecting requests>=2.26
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/ca/91/6d9b8ccacd0412c08820f72cebaa4f0c0441b5cda699c90f618b6f8a1b42/requests-2.28.1-py3-none-any.whl (62 kB)
Collecting markdown>=3.2
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/86/be/ad281f7a3686b38dd8a307fa33210cdf2130404dfef668a37a4166d737ca/Markdown-3.4.1-py3-none-any.whl (93 kB)
Collecting pygments>=2.12
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/4f/82/672cd382e5b39ab1cd422a672382f08a1fb3d08d9e0c0f3707f33a52063b/Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Collecting jinja2>=3.0.2
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/bc/c3/f068337a370801f372f2f8f6bad74a5c140f6fda3d9de154052708dd3c65/Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting mkdocs>=1.4.0
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/ff/00/58f2939f8e6c5f981d9ad9b685c9915a3b315c8f34ba93f0020d64929f70/mkdocs-1.4.2-py3-none-any.whl (3.7 MB)
Collecting mkdocs-material-extensions>=1.1
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/c3/2a/023d300cdbfb0993ddf356d7b02a12abd6c16fbaf72e4411d6aa6233d046/mkdocs_material_extensions-1.1-py3-none-any.whl (7.9 kB)
Collecting pymdown-extensions>=9.4
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/33/85/397e718b306f4a8a8743f31d1c1a79679711d1d52bcd6224dd9ff6d8082c/pymdown_extensions-9.8-py3-none-any.whl (218 kB)
Collecting MarkupSafe>=2.0
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/2c/81/91062a81ac8a18f557f12e2618475b53878755c016c9914c8aa207155c4e/MarkupSafe-2.1.1-cp38-cp38-win_amd64.whl (17 kB)
Collecting importlib-metadata>=4.4
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/b5/64/ef29a63cf08f047bb7fb22ab0f1f774b87eed0bb46d067a5a524798a4af8/importlib_metadata-5.0.0-py3-none-any.whl (21 kB)
Collecting ghp-import>=1.0
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/f7/ec/67fbef5d497f86283db54c22eec6f6140243aae73265799baaaa19cd17fb/ghp_import-2.1.0-py3-none-any.whl (11 kB)
Collecting watchdog>=2.0
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/61/3f/a2fc9452b8161862a78674fda583d8e1addbbecf57a1e172b4a7a96e8087/watchdog-2.1.9-py3-none-win_amd64.whl (78 kB)
Collecting pyyaml-env-tag>=0.1
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/5a/66/bbb1dd374f5c870f59c5bb1db0e18cbe7fa739415a24cbd95b2d1f5ae0c4/pyyaml_env_tag-0.1-py3-none-any.whl (3.9 kB)
Collecting markdown>=3.2
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/f3/df/ca72f352e15b6f8ce32b74af029f1189abffb906f7c137501ffe69c98a65/Markdown-3.3.7-py3-none-any.whl (97 kB)
Collecting mergedeep>=1.3.4
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/2c/19/04f9b178c2d8a15b076c8b5140708fa6ffc5601fb6f1e975537072df5b2a/mergedeep-1.3.4-py3-none-any.whl (6.4 kB)
Collecting packaging>=20.5
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/05/8e/8de486cbd03baba4deef4142bd643a3e7bbe954a784dc1bb17142572d127/packaging-21.3-py3-none-any.whl (40 kB)
Collecting colorama>=0.4
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/d1/d6/3965ed04c63042e047cb6a3e6ed1a63a35087b6a609aa3a15ed8ac56c221/colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting click>=7.0
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/c2/f1/df59e28c642d583f7dacffb1e0965d0e00b218e0186d7858ac5233dce840/click-8.1.3-py3-none-any.whl (96 kB)
Collecting pyyaml>=5.1
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/a8/5b/c4d674846ea4b07ee239fbf6010bcc427c4e4552ba5655b446e36b9a40a7/PyYAML-6.0-cp38-cp38-win_amd64.whl (155 kB)
Collecting charset-normalizer<3,>=2
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/db/51/a507c856293ab05cdc1db77ff4bc1268ddd39f29e7dc4919aa497f0adbec/charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/fc/34/3030de6f1370931b9dbb4dad48f6ab1015ab1d32447850b9fc94e60097be/idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/6f/de/5be2e3eed8426f871b170663333a0f627fc2924cc386cd41be065e7ea870/urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting certifi>=2017.4.17
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/1d/38/fa96a426e0c0e68aabc68e896584b83ad1eec779265a028e156ce509630e/certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting python-dateutil>=2.8.1
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/36/7a/87837f39d0296e723bb9b62bbb257d0355c7f6128853c78955f57342a56d/python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting zipp>=0.5
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/40/8a/d63273ed0fa4a3d06f77e7b043f6577d8894e95515b0c187c52e2c0efabb/zipp-3.10.0-py3-none-any.whl (6.2 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/6c/10/a7d0fa5baea8fe7b50f448ab742f26f52b80bfca85ac2be9d35cdd9a3246/pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting six>=1.5
  Using cached https://artifactory.airbus.corp/artifactory/api/pypi/r-airbus-pypi-virtual/packages/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: zipp, watchdog, urllib3, six, pyyaml, pyparsing, pygments, mkdocs-material-extensions, mergedeep, MarkupSafe, idna, colorama, charset-normalizer, certifi, requests, pyyaml-env-tag, python-dateutil, packaging, jinja2, importlib-metadata, click, markdown, ghp-import, pymdown-extensions, mkdocs, mkdocs-material
Successfully installed MarkupSafe-2.1.1 certifi-2022.9.24 charset-normalizer-2.1.1 click-8.1.3 colorama-0.4.6 ghp-import-2.1.0 idna-3.4 importlib-metadata-5.0.0 jinja2-3.1.2 markdown-3.3.7 mergedeep-1.3.4 mkdocs-1.4.2 mkdocs-material-8.5.10 mkdocs-material-extensions-1.1 packaging-21.3 pygments-2.13.0 pymdown-extensions-9.8 pyparsing-3.0.9 python-dateutil-2.8.2 pyyaml-6.0 pyyaml-env-tag-0.1 requests-2.28.1 six-1.16.0 urllib3-1.26.12 watchdog-2.1.9 zipp-3.10.0

(mkdocs_p38) C:\Users\<my_login>\Desktop\p38>mkdocs new .
INFO     -  Project already exists.

(mkdocs_p38) C:\Users\<my_login>\Desktop\p38>echo "theme:\n name: material" >> mkdocs.yml

(mkdocs_p38) C:\Users\<my_login>\Desktop\p38>more mkdocs.yml
site_name: My Docs
"theme:\n name: material"

(mkdocs_p38) C:\Users\<my_login>\Desktop\p38>more mkdocs.yml
site_name: My Docs
theme:
 name: material

(mkdocs_p38) C:\Users\<my_login>\Desktop\p38>mkdocs build
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: C:\Users\<my_login>\Desktop\p38\site
INFO     -  Documentation built in 0.36 seconds

```

**Reply by MathieuAibt** [reply]
Updates of previous answer:

- Using python 3.8, I applied the same process again BUT without deleting `setuptools` files
--> OK/SUCCESS
- Applying the same successful process BUT using python 3.9
--> KO/FAILURE

@squidfunk On your side, how is your process handling python 3.9 ?

**Reply by MathieuAibt** [reply]
Okay...Latest update here...
Hoping this could help...

# TL; DR;
From Python 3.8 to 3.11: trouble with Python 3.9 only

# Experiments
I like using one yml file as the environment definition file. I feel that it wraps nicely pretty much everything you want to put in one environment:
- from the python version you want to use,
- to the packages you want to install
- the channel you would like specific packages to be installed with
- and it also allows for packages you want installed with pip to be installed as so.

Then, I decided to create a very basic `environment.yml` file and test several version of python.
The `environment.yml` "template" was desgined as so:
```
name: mkdocs_only_from_yml_p<PYTHON_VERSION_without_dots>
channels:
    - defaults
dependencies:
    - python=<PYTHON_VERSION>
    - pip
    - pip:
        - <local_dir>\mkdocs_material-8.5.10-py3-none-any.whl
```

# Process description
From there, I used the same process as previously described:
- creating conda environment using `conda env create --file <my_project_dir>\environment.yml`
- activate this environment : `conda activate <my_env_name>`
_Nota : my_env_name is defined in environment.yml's first line: `name: mkdocs_only_from_yml_p<PYTHON_VERSION_without_dots>`_
- initialize mkdocs files using `mkdocs new .`
- filling `mkdocs.yml` file with following content:
```
site_name: My Docs

theme:
  name: material
```
- apply `mkdocs build`

# Results
I tested python versions from 3.8 to 3.11. Here are the results :
| Python version | Status |
|-----------------|--------|
| 3.8 | SUCCESS |
| 3.9 | **FAILURE** |
| 3.10 | SUCCESS |
| 3.11 | SUCCESS |
